### PR TITLE
Rodrigo/create profile

### PR DESCRIPTION
**Pull request by rodtai**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -7,8 +7,11 @@
     "query-string": "^6.7.0",
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
+    "react-dropzone": "^10.1.5",
+    "react-forms-validator": "^1.0.8",
     "react-router-dom": "^5.0.1",
-    "react-scripts": "3.0.1"
+    "react-scripts": "3.0.1",
+    "superagent": "^5.1.0"
```
Comment (gnaww) [inline frontend/package.json]: what is this dependency for?

File: frontend/src/components/DragAndDrop.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react'
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: newline before class

File: frontend/src/components/DragAndDrop.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+class DragAndDrop extends Component {
+    state = {
+        drag: false
+    }
+    dropRef = React.createRef()
+    handleDrag = (e) => {
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: newline before each function to split them up visually

File: frontend/src/components/DragAndDrop.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                                fontSize: 36
+                            }}
+                        >
+                            <div>drop here :)</div>
+                        </div>
+                    </div>
+                }
+                {this.props.children}
+            </div>
+        )
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: semicolon

File: frontend/src/components/DragAndDrop.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                                position: 'absolute',
+                                top: '50%',
+                                right: 0,
+                                left: 0,
+                                textAlign: 'center',
+                                color: 'grey',
+                                fontSize: 36
+                            }}
+                        >
+                            <div>drop here :)</div>
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: can you change the message here?

File: frontend/src/components/DragAndDrop.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                        </div>
+                    </div>
+                }
+                {this.props.children}
+            </div>
+        )
+    }
+}
+export default DragAndDrop;
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: need newline at end of file

File: frontend/src/pages/CreateProfile/BasicInfo.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    componentDidMount() {
+        if (this.props.savedData !== null) {
+            this.setState(this.props.savedData);
+            this.errors = false;
+        }
+    }
+
+    handleFirstNameChange = event => {
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/BasicInfo.js]: you don't have to do this, but to make it cleaner you can refactor all these functions that do the exact same thing into one:
```
handleChange = inputField => event => {
    this.setState({ [inputField]: event.target.value });
}
```

then for example, you could replace your onChange to `onChange={this.handleChange('lastName')}`

File: frontend/src/components/DragAndDrop.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react'
+class DragAndDrop extends Component {
+    state = {
```
Comment (gnaww) [inline frontend/src/components/DragAndDrop.js]: can you change this state to match the other components syntax?

File: frontend/src/pages/CreateProfile/BasicInfo.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                    reference={{ website: this.state.website }}
+                    validationRules={this.state.website.length === 0 ? { required: false } : { url: true }}
+                    validationMessages={this.state.website.length === 0 ? { required: false } : { url: "Not a valid website URL." }}
+                />
+            </div>
+        )
+    }
+}
+
+export default BasicInfo;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/BasicInfo.js]: newline after

File: frontend/src/pages/CreateProfile/CreateProfile.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import styles from '../../styles/CreateProfile.module.css'
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/CreateProfile.js]: semicolon

File: frontend/src/pages/CreateProfile/Constants.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
+export const AnswerTypes = {
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/Constants.js]: missing semicolons after the consts

File: frontend/src/pages/CreateProfile/CreateProfile.js [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+        event.preventDefault();
+        this.setState({ page: this.state.page - 1 });
+    }
+
+    handleNext = event => {
+        event.preventDefault();
+        this.setState({ clickedNext: true });
+    }
+
+    handleFinish = event => {
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/CreateProfile.js]: this seems like it is unused

File: frontend/src/pages/CreateProfile/CreateProfile.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        event.preventDefault();
+    }
+
+    handleOnSubmitData = (data, errors) => {
+        if (!errors) {
+            var nextPage = this.state.page + 1;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/CreateProfile.js]: can you change all instances of `var` to `let` to follow ES6 syntax

File: frontend/src/pages/CreateProfile/OptionalQuestions.js [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+                    questions.map((qa, idx) => {
+                        return getTextboxQuestion(qa, this.setTextboxValue, this.state, idx)
+                    })
+                }
+            </div>
+        )
+    }
+}
+
+export default OptionalQuestions;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/OptionalQuestions.js]: newline

File: frontend/src/pages/CreateProfile/Options.js [hunk 15]
```diff
@@ -0,0 +1,4 @@
+export var roleTypes = {
+    practitioner: "practitioners",
+    researcher: "researcher"
+}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/Options.js]: newline

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+            researchTopics: getCheckedValuesArray(PractitionerInformation.ResearchTopics)
+        }
+    }
+
+    componentDidUpdate(_prevProps, _prevState) {
+        function keyIsValid(state, key) {
+            var last = state[key].length - 1;
+            return state[key].filter(r => r.checked).length > 0
+                && (!state[key][last].checked || state[key][last].other !== "");
+        }
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: newline

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+        function isInvalid(state) {
+            var locationValid = state.location.instituition !== "" && state.location.address !== "";
+            var keys = ["roleDescriptions", "ageGroups", "deliveryTypes", "researchTopics"];
+            var validArray = keys.map(k => keyIsValid(state, k));
+            console.log(validArray);
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: remove this

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+        if (this.props.savedData !== null) {
+            this.setState(this.props.savedData);
+        }
+    }
+
+    setLocatedAtCornell = event => {
+        this.setState({
+            locatedAtCornell: event.target.value,
+            location: {
+                instituition: "",
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: institution not instituition

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+                address: ""
+            }
+        });
+    }
+
+    setLocationDropdown = event => {
+        this.setState({
+            location: {
+                instituition: "Cornell Cooperative Extension",
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: institution not instituition

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+                                this.state.locatedAtCornell === "true" &&
+                                getDropDownQuestion(qa.extra, this.setLocationDropdown)
+                            }
+                            {
+                                this.state.locatedAtCornell === "false" &&
+                                (
+                                    <>
+                                        <input
+                                            className={styles.longTextInput}
+                                            placeholder="What is your instituition or organization?"
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: institution not instituition

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 21]
```diff
@@ -0,0 +1,149 @@
...
+                                            type="text"
+                                            value={this.state.location.instituition}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: institution not instituition

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+                                            onChange={this.setLocationTextbox("instituition")}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: institution not instituition

File: frontend/src/pages/CreateProfile/PractitionerQuestions.js [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+                            return this.getQAComponent(qa, index)
+                        })
+                    }
+                </ol>
+            </div>
+        )
+    }
+}
+
+export default PractitionerQuestions;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/PractitionerQuestions.js]: newline

File: frontend/src/pages/CreateProfile/QAComponents.js [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+                                    onChange={e => handlerFunction(qa.answer.key, option, e.target.value)}
+                                />
+                            )}
+                        </div>
+                    )
+                })
+            }
+        </>
+    )
+}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/QAComponents.js]: newline

File: frontend/src/pages/CreateProfile/ResearcherQuestions.js [hunk 25]
```diff
@@ -0,0 +1,262 @@
...
+                            return this.getQAComponent(qa, index)
+                        })
+                    }
+                </ol>
+            </div>
+        )
+    }
+}
+
+export default ResearcherQuestions;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/ResearcherQuestions.js]: newline

File: frontend/src/pages/CreateProfile/ReviewFinish.js [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+                            </div>
+                        )
+                    })
+                }
+            </div>
+        )
+    }
+}
+
+export default ReviewFinish;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/ReviewFinish.js]: newline

File: frontend/src/pages/CreateProfile/ReviewFinish.js [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+                {
+                    navigationOptions.map((nav, index) => {
+                        return (
+                            <div className={styles.roleCard} key={index}>
+                                <Link to={nav.link}>
+                                    <img
+                                        className={nav.unclicked}
+                                        src={nav.img}
+                                        onClick={this.handleClick(index)}
+                                        alt={"TODO"}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/ReviewFinish.js]: TODO?

File: frontend/src/pages/CreateProfile/RoleSelection.js [hunk 28]
```diff
@@ -0,0 +1,78 @@
...
+            <div className={styles.rolePage}>
+                {
+                    roleOptions.map((role, index) => {
+                        return (
+                            <div className={styles.roleCard} key={index}>
+                                <img
+                                    className={this.state.clicked === index ? role.clicked : role.unclicked}
+                                    src={role.img}
+                                    onClick={() => this.handleSelectRole(index, role.value)}
+                                    alt={"TODO"}
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/RoleSelection.js]: TODO?

File: frontend/src/pages/CreateProfile/RoleSelection.js [hunk 29]
```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import styles from '../../styles/CreateProfile.module.css';
+import researcher from '../../images/researcher.svg';
+import practitioner from '../../images/practitioner.svg';
+import { roleTypes } from './Options'
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/RoleSelection.js]: semicolon

File: frontend/src/pages/CreateProfile/RoleSelection.js [hunk 30]
```diff
@@ -0,0 +1,78 @@
...
+        clicked: styles.researcherImgClicked
+    },
+    {
+        text: "4-H Educator, other Cornell Cooperative Extension (CCE) Role, or other primarily practice-focused role.",
+        value: roleTypes.practitioner,
+        img: practitioner,
+        unclicked: styles.practitionerImg,
+        clicked: styles.practitionerImgClicked
+    }
+
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/RoleSelection.js]: extra newline

File: frontend/src/pages/CreateProfile/RoleSelection.js [hunk 31]
```diff
@@ -0,0 +1,78 @@
...
+                            </div>
+                        )
+                    })
+                }
+            </div>
+        )
+    }
+}
+
+export default RoleSelection;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/RoleSelection.js]: newline

File: frontend/src/pages/CreateProfile/UploadProPic.js [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+                        <label className={styles.imageUpload} for="propic">SELECT FILE TO UPLOAD</label>
+                        <input className={styles.propic} onChange={this.setProfilePicture} type="file" id="propic" accept="image/*" />
+                    </div>
+                </DragAndDrop>
+            </div>
+        )
+    }
+}
+
+export default UploadProPic;
```
Comment (gnaww) [inline frontend/src/pages/CreateProfile/UploadProPic.js]: newline

File: frontend/src/styles/CreateProfile.module.css [hunk 33]
```diff
@@ -0,0 +1,304 @@
...
+.studiesNavImg {
+    composes: researcherImg;
+    width: 251px;
+    height: 201px;
+    padding: 53px 21px 37px 21px;
+}
+
+.finishPage {
+    display: flex;
+}
```
Comment (gnaww) [inline frontend/src/styles/CreateProfile.module.css]: newline

File: frontend/package.json (hunk 1 above)
Comment (rodtai) [reply frontend/package.json]: Must be an old dependency that I never uninstalled.

File: frontend/src/pages/CreateProfile/ReviewFinish.js (hunk 27 above)
Comment (rodtai) [reply frontend/src/pages/CreateProfile/ReviewFinish.js]: No idea what to put for this.

File: frontend/src/pages/CreateProfile/RoleSelection.js (hunk 28 above)
Comment (rodtai) [reply frontend/src/pages/CreateProfile/RoleSelection.js]: No idea what to put for this.

**Comment by gnaww** [comment]
Closes https://github.com/gnaww/pryde-website/issues/5

**Pull request merged**